### PR TITLE
Skip notary tests that break --net none build

### LINKED ISSUE #15049: notary tests break --net none build
fix the --net none failures seen here

https://jenkins.dockerproject.org/job/Docker%20Master%20(--net=none)/1296/console

### PR DESCRIPTION
**Pull request by boydjd**
Fixes #15049

Signed-off-by: Joshua Boyd boydjd@jbip.net

### CODE REVIEW

**Comment by runcom** [comment]
LGTM if janky's happy

**Comment by NathanMcCauley** [comment]
If its needed here then `testRequires(c, Network)` is probably needed on docker_cli_{push,create,build}_test.go as well.

**Comment by tiborvass** [comment]
@boydjd would you mind updating according to what Nathan said? Thanks for your PR!

**Comment by boydjd** [comment]
You want `testRequires(c, Network)` added to every test in docker_cli_{push,create,build}_test.go?

**Comment by tiborvass** [comment]
@boydjd i think he means only the other tests that are based on notary. You can have a look at #14546 to see which tests it is. Thank you!

**Comment by NathanMcCauley** [comment]
@boydjd Why was `push` failing? I didn't notice it failing locally and I don't see why it would've been different than any of the other notary tests we added that talk to a local registry and notary server.

**Comment by boydjd** [comment]
@NathanMcCauley 

```
FAIL: docker_cli_pull_test.go:159: DockerTrustSuite.TestTrustedPull

docker_cli_pull_test.go:167:
    c.Fatalf("Error running trusted pull: %s\n%s", err, out)
... Error: Error running trusted pull: exit status 1
could not validate the path to a trusted root: unable to retrieve valid leaf certificates
```

**Comment by tiborvass** [comment]
Ping @NathanMcCauley @diogomonica @dmcgowan

**Comment by cpuguy83** [comment]
ping @NathanMcCauley

**Comment by dmcgowan** [comment]
`TestTrustedPull` does not make network calls, the notary server is running locally. When I tested with `--net none` this test still passes.

`TestRunContainerWithBridgeNone` I don't understand this test but noticed it does pass with this change and not before :smile: Although it is not related to Notary.

File: integration-cli/docker_cli_daemon_test.go [hunk 1]
```diff
@@ -1406,6 +1406,7 @@ func (s *DockerDaemonSuite) TestCleanupMountsAfterCrash(c *check.C) {
 
 func (s *DockerDaemonSuite) TestRunContainerWithBridgeNone(c *check.C) {
 	testRequires(c, NativeExecDriver)
+	testRequires(c, Network)
```
Comment (jfrazelle) [inline integration-cli/docker_cli_daemon_test.go]: remove this one please

File: integration-cli/docker_cli_pull_test.go [hunk 2]
```diff
@@ -157,6 +157,8 @@ func (s *DockerSuite) TestPullImageWithAllTagFromCentralRegistry(c *check.C) {
 }
 
 func (s *DockerTrustSuite) TestTrustedPull(c *check.C) {
+	testRequires(c, Network)
```
Comment (jfrazelle) [inline integration-cli/docker_cli_pull_test.go]: we dont need this

**Comment by jfrazelle** [comment]
sorry after further review these should not be skipped, if they are breaking net none its for other reasons.

**Comment by jfrazelle** [comment]
also i believe the networking team had ideas on the bridge problems w that test

**Pull request closed without merging**